src/common/io/tls/session.c

```diff
@@ -144,77 +184,33 @@ tlsSessionRead(THIS_VOID, Buffer *buffer, bool block)
 
     ssize_t result = 0;
 
+    // If no TLS data pending then check the socket to reduce blocking
+    if (!SSL_pending(this->session))
+        sckSessionReadyRead(this->socketSession);
+
     // If blocking read keep reading until buffer is full
     do
     {
-        // If no tls data pending then check the socket
-        if (!SSL_pending(this->session))
-            sckSessionReadyRead(this->socketSession);
-
         // Read and handle errors
-        result = SSL_read(this->session, bufRemainsPtr(buffer), (int)bufRemains(buffer));
+        result = tlsSessionResult(this, (int)SSL_read(this->session, bufRemainsPtr(buffer), (int)bufRemains(buffer)), true);
```

Comment:
Why did you feel it necessary to cast the SSL_read result to an (int) - the function is declared as returning an int and I see no where else in the code where a call to tlsSessionResult from an SSL_write, SSL_connect, SSL_accept (all returning an int) needed a cast, so why here?

Reply:
Hmmm, not sure. This may have been copied from `IoHandleWrite` which is using `ssize_t` or maybe I wanted to make it look more like size_t which is used by `Buffer`. 

Anyway, all that is useless so I have changed result to `int` and removed the cast.

Reply:
Yes, and `ssize_t result` was changed to `int result` since that's what is actually returned.